### PR TITLE
Issue491

### PR DESCRIPTION
**Pull request by oxarbitrage**
https://github.com/bitshares/bitshares-core/issues/491

### CODE REVIEW

File: libraries/chain/market_evaluator.cpp [hunk 1]
```diff
@@ -189,6 +175,7 @@ void_result call_order_update_evaluator::do_apply(const call_order_update_operat
       {
          d.modify(_paying_account->statistics(d), [&](account_statistics_object& stats) {
                stats.total_core_in_orders += o.delta_collateral.amount;
+               assert(dynamic_asset.current_supply <= 0);
```
Comment (abitmore) [inline libraries/chain/market_evaluator.cpp]: This is apparently wrong.
By the way don't use `assert()`, see https://github.com/bitshares/bitshares-core/issues/511

**Comment by oxarbitrage** [comment]
i re added the `delta_collateral` assert back as i am not sure if it is safe to remove.

**Comment by abitmore** [comment]
Perhaps better rebase/squash the commits.

**Comment by oxarbitrage** [comment]
it can be done at merge time(the squash) into 1 commit if you approve the changes.

**Pull request merged**